frontend/model/contracts/group.js

```diff
@@ -141,6 +142,47 @@ function isActionYoungerThanUser (actionMeta: Object, userProfile: ?Object): boo
   return compareISOTimestamps(actionMeta.createdDate, userProfile.joinedDate) > 0
 }
 
+function updateGroupStreaks ({ state, getters }) {
+  const streaks = state.streaks
+  const cPeriod = getters.groupSettings.distributionDate
+  const thisPeriodPayments = getters.paymentsForPeriod(cPeriod)
+  const thisPeriodDistribution = adjustedDistribution({
+    distribution: unadjustedDistribution({
+      haveNeeds: getters.haveNeedsForThisPeriod(cPeriod),
+      minimize: getters.groupSettings.minimizeDistribution
+    }) || [],
+    payments: getters.paymentsForPeriod(cPeriod),
+    dueOn: getters.dueDateForPeriod(cPeriod)
+  })
```

Comment:
@taoeffect does sound good. 👍 

> Have you tested this functionality in the beginning of a 2nd payment period by setting up a group and adjusting your computer's clock?

No I actually repeatedly called `sbp('gi.actions/group/forceDistributionDate', { contractID: sbp('state/vuex/state').currentGroupId })` in dev console for testing this functionality, as I added this function call in there for dev purpose.

Reply:
@SebinSong OK, well, I would try testing by adjusting the computer clock, because a lot of code uses `periodBeforePeriod` and `periodAfterPeriod` getters - and those won't work with `forceDistributionDate`, because they work by adding 30 days to the period, so they would skip over the periods that were created because of `forceDistributionDate`

Reply:
@taoeffect Thanks for providing more context, and yeah I did test it via adjusting the clock of my computer and it did seem to work well for me. please go ahead with reviewing that way. hope it works for you too.